arelle/ModelDocument.py

```diff
@@ -1383,7 +1384,7 @@ def testcaseDiscover(self, testcaseElement):
         if XmlUtil.xmlnsprefix(testcaseElement, XbrlConst.cfcn) or isTransformTestcase:
             self.type = Type.REGISTRYTESTCASE
         self.outpath = self.xmlRootElement.get("outpath") or self.filepathdir
-        self.testcaseVariations = []
+        self.testcaseVariations: list[ModelTestcaseVariation] = []
```

Comment:
This and the change in `validation_util` was added as I was trying to debug the failing tests. I hope these small changes can remain in this PR.